huaweicloud/services/fgs/data_source_huaweicloud_fgs_functions.go

```diff
@@ -133,134 +158,178 @@ func DataSourceFunctionGraphFunctions() *schema.Resource {
 						},
 						"log_stream_id": {
-							Type:     schema.TypeString,
-							Computed: true,
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The LTS log stream ID.`,
 						},
 						"functiongraph_version": {
-							Type:     schema.TypeString,
-							Computed: true,
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The functionGraph version.`,
 						},
 					},
 				},
+				Description: `All functions that match the filter parameters.`,
 			},
 		},
 	}
 }
 
-func dataSourceFunctionGraphFunctionsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	conf := meta.(*config.Config)
-	region := conf.GetRegion(d)
-
-	fgsClient, err := conf.FgsV2Client(region)
-	if err != nil {
-		return diag.Errorf("error creating FunctionGraph V2 client: %s", err)
+func buildFunctionsQueryParams(d *schema.ResourceData) string {
+	if pkgName, ok := d.GetOk("package_name"); ok {
+		return fmt.Sprintf("&package_name=%v", pkgName)
 	}
+	return ""
+}
 
-	// MaxItems and Marker use default values.
-	opts := function.ListOpts{
-		PackageName: d.Get("package_name").(string),
-	}
-	allPages, err := function.List(fgsClient, opts).AllPages()
-	if err != nil {
-		return diag.Errorf("error querying functions: %s", err)
-	}
-	resp, err := function.ExtractList(allPages)
-	if err != nil {
-		return diag.Errorf("error querying functions: %s", err)
-	}
+func getFunctions(client *golangsdk.ServiceClient, d *schema.ResourceData) ([]interface{}, error) {
+	var (
+		httpUrl = "v2/{project_id}/fgs/functions?maxitems=100"
+		marker  = 0
+	)
 
-	filterResult, err := filterFunctionList(d, resp.Functions, conf)
-	if err != nil {
-		return diag.FromErr(err)
-	}
-	uuid, err := uuid.GenerateUUID()
-	if err != nil {
-		return diag.Errorf("unable to generate ID: %s", err)
+	listPath := client.Endpoint + httpUrl
+	listPath = strings.ReplaceAll(listPath, "{project_id}", client.ProjectID)
+	listPath += buildFunctionsQueryParams(d)
+	listOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
 	}
-	d.SetId(uuid)
-	mErr := multierror.Append(nil,
-		d.Set("region", region),
-		d.Set("functions", flattenFunctions(filterResult)),
-	)
 
-	if err := mErr.ErrorOrNil(); err != nil {
-		return diag.Errorf("error saving datas of FunctionGraph functions: %s", err)
+	result := make([]interface{}, 0)
+	for {
+		listPathWithMarker := fmt.Sprintf("%s&marker=%v", listPath, marker)
+		requestResp, err := client.Request("GET", listPathWithMarker, &listOpt)
+		if err != nil {
+			return nil, err
+		}
+		respBody, err := utils.FlattenResponse(requestResp)
+		if err != nil {
+			return nil, err
+		}
+		functions := utils.PathSearch("functions", respBody, make([]interface{}, 0)).([]interface{})
+		if len(functions) < 1 {
+			break
+		}
+		result = append(result, functions...)
+		marker += len(functions)
```

Comment:
Marker represents the last record position queried last time, it is not offset.

Reply:
done.